pkg/util/labels.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+// GenerateOutputImageLabels generate the labels based on the source repository
+// informations.
+func GenerateOutputImageLabels(info *api.SourceInfo, config *api.Config) map[string]string {
+	result := map[string]string{}
+
+	if len(config.ContextDir) > 0 {
+		result[api.DefaultNamespace+"build.source-context-dir"] = config.ContextDir
+	}
+
+	if len(config.Description) > 0 {
+		result[api.KubernetesNamespace+"description"] = config.Description
+	}
+
+	if len(config.DisplayName) > 0 {
+		result[api.KubernetesNamespace+"display-name"] = config.DisplayName
+	} else {
+		result[api.KubernetesNamespace+"display-name"] = config.Tag
+	}
+
+	if info == nil {
+		glog.V(5).Infof("No source information!")
+		return result
+	}
+
+	result[api.DefaultNamespace+"build.commit.author"] = info.Author
+	result[api.DefaultNamespace+"build.commit.date"] = info.Date
+	result[api.DefaultNamespace+"build.commit.id"] = info.CommitID
+	result[api.DefaultNamespace+"build.commit.ref"] = info.Ref
+	result[api.DefaultNamespace+"build.commit.message"] = info.Message
+	result[api.DefaultNamespace+"build.source-location"] = info.Location
```

Comment:
might be better to check that you have a value for each field.  in the future we might support an SCM that has a subset of these fields.